src/csharp/Grpc.Core/Internal/CallSafeHandle.cs

```diff
@@ -218,5 +218,16 @@ private static uint GetFlags(bool buffered)
         {
             return buffered ? 0 : GRPC_WRITE_BUFFER_HINT;
         }
+
+        /// <summary>
+        /// Only for testing.
+        /// </summary>
+        public static CallSafeHandle CreateFake(IntPtr ptr, CompletionQueueSafeHandle cq)
+        {
+            var call = new CallSafeHandle();
+            call.SetHandle(ptr);
```

Comment:
If I don't set a fake pointer, the safehandle's handle would be "invalid" (= null) and that could have different marshalling behavior than an actual pointer.